src/commands/prepare.ts

```diff
@@ -497,30 +463,39 @@ export async function releaseMain(argv: ReleaseOptions): Promise<any> {
     githubConfig.repo
   );
   logger.debug(`Default branch for the repo:`, defaultBranch);
+  const repoStatus = await git.status();
+  const rev = argv.rev || repoStatus.current || defaultBranch;
 
-  // Check that we're in an acceptable state for the release
-  await checkGitState(git, defaultBranch, !argv.noGitChecks);
-
-  // Check whether the version/tag already exists
-  await checkForExistingTag(git, newVersion, !argv.noGitChecks);
+  if (argv.noGitChecks) {
+    logger.info('Not checking the status of the local repository');
+  } else {
+    // Check that we're in an acceptable state for the release
+    checkGitStatus(repoStatus, rev);
+  }
 
-  // Check & update the changelog
-  await prepareChangelog(
-    newVersion,
-    argv.noChangelog ? ChangelogPolicy.None : config.changelogPolicy,
-    config.changelog
-  );
+  logger.info(`Releasing version ${newVersion} from ${rev}`);
+  if (!argv.rev && rev !== defaultBranch) {
+    await promptConfirmation();
+  }
 
   logger.info(`Preparing to release the version: ${newVersion}`);
 
   // Create a new release branch and check it out. Throw an error if it already
   // exists.
   const branchName = await createReleaseBranch(
     git,
+    rev,
     newVersion,
     config.releaseBranchPrefix
   );
 
+  // Check & update the changelog
+  await prepareChangelog(
```

Comment:
This step should have come _after_ creating the release branch from the beginning. Old time bug fixed :)